toolkit/src/components/cut/list-item/index.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { assert } from '@ember/debug';
+
+export interface ComponentSignature {
+  Args: {
+    href?: string;
+    isHrefExternal?: boolean;
+    route?: string;
+    isRouteExternal?: boolean;
+    query?: object;
+    replace?: string;
+    onClick(): void;
+  };
+}
+
+export default class ListItemComponent extends Component<ComponentSignature> {
+  get route() {
+    const { onClick, route, href } = this.args;
+
+    if (route !== undefined) {
+      assert(
+        `Cut::ListItem" could have applied either @route, @href or @onClick.`,
+        onClick === undefined && href === undefined
+      );
+    }
+
+    return route;
+  }
+
+  get href() {
+    const { onClick, route, href } = this.args;
+
+    if (href !== undefined) {
+      assert(
+        `Cut::ListItem" could have applied either @href, @route or @onClick.`,
+        onClick === undefined && route === undefined
+      );
+    }
+    return href;
+  }
+
+  @action
+  onClickAction() {
+    this.args.onClick?.();
+  }
```

Comment:
I _think_ there is something not working with the logic/implementation of the `onClick` use case: try to add a console.log here, and click on the third item in the demo page ("Element with onClick")
```suggestion
  onClickAction() {
    console.log('onClickAction clicked!');
    this.args.onClick?.();
  }
```
you will not see in the devtools console when you click the item.